lib/smack/Prelude.cpp

```diff
@@ -326,37 +655,65 @@ std::string Prelude::getPrelude() {
     s << Decl::typee(Naming::UNINTERPRETED_FLOAT_TYPE, "") << "\n";
   }
   s << "\n";
+}
 
+void Prelude::generatePtrConstant(unsigned val, std::stringstream& s) {
+  std::string ptrConst = indexedName("$"+std::to_string(val), {Naming::PTR_TYPE});
+  s << Decl::constant(ptrConst, Naming::PTR_TYPE) << "\n";
+  s << Decl::axiom(Expr::eq(Expr::id(ptrConst), rep.pointerLit((unsigned long long)val))) << "\n";
+}
+
+void Prelude::generateIntConstant(unsigned val, std::stringstream& s) {
+  std::string intConst = "$"+std::to_string(val);
+  s << Decl::constant(intConst,rep.intType(32)) << "\n";
+  s << Decl::axiom(Expr::eq(Expr::id(intConst),rep.integerLit((unsigned long long)val,32))) << "\n";
+}
+
+void Prelude::generateConstants(std::stringstream& s) {
   s << "// Basic constants" << "\n";
-  s << Decl::constant("$0",rep.intType(32)) << "\n";
-  s << Decl::axiom(Expr::eq(Expr::id("$0"),rep.integerLit(0ULL,32))) << "\n";
-  s << Decl::constant("$1",rep.intType(32)) << "\n";
-  s << Decl::axiom(Expr::eq(Expr::id("$1"),rep.integerLit(1ULL,32))) << "\n";
+  generateIntConstant(0, s);
+  generateIntConstant(1, s);
 
+  // we don't need so many constants
```

Comment:
Why are you saying this? I suspect we introduced them one by one whenever we needed one. No?